### PR TITLE
cleanup(GCS+gRPC): add public ReadObjectRequest

### PR DESCRIPTION
**Pull request by coryan**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/12604)

### CODE REVIEW

**Review by scotthart — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/12604)* status: 0 of 5 files reviewed, all discussions resolved

___
*[`google/cloud/storage/async_object_requests.h` line 147 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/12604#-NdvvLWsCPYsKRTJWgXb:-NdvvLWsCPYsKRTJWgXc:bp5je3s) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/d92dafdc88fb65abdfcd51988e73cf37606fd0c4/google/cloud/storage/async_object_requests.h#L147)):*
> ```objc
>   }
>   template <typename O>
>   O GetOption() const {
> ```

While `T` is also a single character, it's weird seeing just an `O`. Maybe because `T` is so ubiquitous or maybe at first glance I wondered if this was a `0`.

File: google/cloud/storage/async_object_requests.h [hunk 1]
```diff
@@ -111,6 +111,48 @@ class InsertObjectRequest {
   Impl impl_;
 };
 
+/**
+ * A request to read an object.
+ *
+ * This class can hold all the mandatory and optional parameters to read an
+ * object.
+ *
+ * This class is the public API for the library because it is required for
```
Comment (devbww) [inline google/cloud/storage/async_object_requests.h:120]: Perhaps all of these should read *"is **in** the public API"*.

Comment (coryan) [reply google/cloud/storage/async_object_requests.h]: Done, here and in the other classes.

**Review by coryan — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/12604)* status: 0 of 5 files reviewed, 1 unresolved discussion (waiting on @devbww and @scotthart)

___
*[`google/cloud/storage/async_object_requests.h` line 147 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/12604#-NdvvLWsCPYsKRTJWgXb:-NdzVRgo1fAAtzVDA5C2:bk0uz19) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/d92dafdc88fb65abdfcd51988e73cf37606fd0c4/google/cloud/storage/async_object_requests.h#L147)):*
<details><summary><i>Previously, scotthart (Scott Hart) wrote…</i></summary><blockquote>

While `T` is also a single character, it's weird seeing just an `O`. Maybe because `T` is so ubiquitous or maybe at first glance I wondered if this was a `0`.
</blockquote></details>

Done, here and in the existing classes

File: google/cloud/storage/async_object_requests.h [hunk 2]
```diff
@@ -72,7 +72,7 @@ class WritePayload {
  * must be in an owning type, such as `WritePayload`. For blocking request, a
  * non-owning type (such as `absl::string_view`) can reduce data copying.
  *
- * This class is the public API for the library because it is required for
+ * This class in in the public API for the library because it is required for
```
Comment (devbww) [inline google/cloud/storage/async_object_requests.h:75]: `s/in in/is in/` ...

Comment (coryan) [reply google/cloud/storage/async_object_requests.h:75]: D'oh.  I sent a new PR to fix this.

**Pull request merged**